sdk/core/core/inc/az_assert.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_ASSERT_H
+#define AZ_ASSERT_H
+
+#include <assert.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AZ_ASSERT(CONDITION) assert(CONDITION)
```

Comment:
Why do we need our own assert macro?

Reply:
1. we would like to have some control what does it do.
2. I plan to change the macro to somithing like `AZ_PRECONDITION`.

Reply:
I see.  Before we double-down on our own implementation we should check out the contract system in azure-ulib-c.  If we don't want/need to depend on that repo in full we can copy what they have.  Of course this would also mean they need to make that repo public.